ORStools/proc/directions_lines_proc.py

```diff
@@ -166,7 +169,7 @@ def processAlgorithm(self, parameters, context, feedback):
         return {self.OUT: dest_id}
 
     @staticmethod
-    def _get_sorted_lines(layer, field_name):
+    def _get_sorted_lines(layer: QgsFeatureSource, field_name: str) -> Generator:
```

Comment:
I think this could also be a `QgsProcessingFeatureSource`, as it operates on the output of `parameterAsSource()`.

Reply:
Yes, printing `source` gives me `QgsProcessingFeatureSource`.

Reply:
Resolved in edcd29fb48b7aa76f6923369ea553148714d731b